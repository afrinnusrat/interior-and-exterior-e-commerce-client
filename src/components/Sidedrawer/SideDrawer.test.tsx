import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import { SideDrawer, SideDrawerProps } from './SideDrawer.component';

afterEach(cleanup);

describe('<SideDrawer />', () => {
  const mockHandleClose = jest.fn();

  let props: SideDrawerProps = {
    toggled: false,
    zIndex: 1,
    title: 'Title',
    handleClose: mockHandleClose,
    children: <span>test</span>
  };

  it('renders correctly', () => {
    const { asFragment } = render(<SideDrawer {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('contains correct style rules in `Wrapper` element', () => {
    const { getByTestId } = render(
      <SideDrawer {...props} toggled={true} zIndex={2} />
    );
    const wrapper = getByTestId('Wrapper');

    expect(wrapper).toHaveStyle('z-index: 2');
    expect(wrapper).toHaveStyle('transform: translateX(0)');
  });

  it('calls `handleClose` func', () => {
    const { getByTestId } = render(<SideDrawer {...props} />);
    const btn = getByTestId('CloseBtn');

    fireEvent.click(btn);
    expect(mockHandleClose).toHaveBeenCalledTimes(1);
  });
});
