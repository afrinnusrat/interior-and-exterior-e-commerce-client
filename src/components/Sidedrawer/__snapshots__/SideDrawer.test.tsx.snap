// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideDrawer /> renders correctly 1`] = `
<DocumentFragment>
  <nav
    class="sc-bdVaJa heZQzu"
    data-testid="Wrapper"
  >
    <header
      class="sc-bwzfXH gEGrQn"
    >
      <p
        class="sc-htpNat cuHHjG"
      >
        Title
      </p>
      <button
        class="sc-bxivhb gkQASD"
        data-testid="CloseBtn"
      >
        <i
          class="far fa-times-circle"
        />
      </button>
    </header>
    <ul
      class="sc-ifAKCX hshtzb"
    >
      <span>
        test
      </span>
    </ul>
  </nav>
</DocumentFragment>
`;
